vars/mavenExecute.groovy

```diff
@@ -9,19 +9,18 @@ import groovy.transform.Field
 
 @Field Set GENERAL_CONFIG_KEYS = []
 @Field Set STEP_CONFIG_KEYS = [
+    'defines',
     'dockerImage',
-    'globalSettingsFile',
-    'projectSettingsFile',
-    'pomPath',
-    'm2Path'
-]
-@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS.plus([
     'dockerOptions',
     'flags',
+    'globalSettingsFile',
     'goals',
-    'defines',
-    'logSuccessfulMavenTransfers'
-])
+    'logSuccessfulMavenTransfers',
+    'm2Path',
+    'pomPath',
+    'projectSettingsFile',
+]
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
```

Comment:
Not sure why goals and flags need to be step configurations, I think these highly depend on context where the step is used. Might be different in each step invokation. Whereas the dockerImage or the settings file is more a general configuration which is common for multiple executions.

Reply:
when they are onla parameter keys they can only be provided via signature when calling the step. With this approach the values are part of the pipeline. Hence it is more complicated to re-use the pipeline either by copying (... bad, bad, bad) or by referencing it from several pipelines. Better to have the parameters provided on the stage level. Since there is currently no separation between step and stage level, the only way is to provide it on step level. Maybe we should introduce another set in order to differenciate between step and stage level configuration ...

This is why I think it is better to have the values not only as parameter keys.